IonDotnet/Tree/IValueFactory.cs

```diff
@@ -0,0 +1,195 @@
+﻿using System;
+using System.Numerics;
+
+namespace IonDotnet.Tree
+{
+    /// <summary>
+    /// The factory for all {@link IonValue}s.
+    /// WARNING: This interface should not be implemented or extended by
```

Comment:
We warned against implementing most of our public interfaces in ion-java and there's now fairly general agreement that that was a mistake. Let's just remove this guidance.

Reply:
Removed.